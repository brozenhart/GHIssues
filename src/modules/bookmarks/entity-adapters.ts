import { createEntityAdapter } from '@reduxjs/toolkit';
import { RootState } from 'root-types';
import { Bookmarks } from './types';

export const bookmarksAdapter = createEntityAdapter<Bookmarks>({
  selectId: data => data.id,
});

export const bookmarksSelectors = bookmarksAdapter.getSelectors<RootState>(
  state => state.bookmarks,
);
