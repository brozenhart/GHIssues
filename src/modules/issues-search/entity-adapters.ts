import { createEntityAdapter } from '@reduxjs/toolkit';
import { RootState } from 'root-types';
import { IssuesResponseData } from './models';

export const issuesAdapter = createEntityAdapter<IssuesResponseData>({
  selectId: data => data.id,
});

export const issuesSelectors = issuesAdapter.getSelectors<RootState>(
  state => state.issuesSearch,
);
