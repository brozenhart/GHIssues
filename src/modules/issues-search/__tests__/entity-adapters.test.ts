import { issuesAdapter, issuesSelectors } from '../entity-adapters';

describe('entity adapters', () => {
  describe('issues adapter', () => {
    it('creates an entity adapter for issues', () => {
      expect(issuesAdapter).toMatchSnapshot();
    });

    it('returns generated selector for the issues adapter', () => {
      expect(issuesSelectors).toMatchSnapshot();
    });
  });
});
