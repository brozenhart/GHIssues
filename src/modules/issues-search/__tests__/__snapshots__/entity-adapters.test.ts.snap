// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entity adapters issues adapter creates an entity adapter for issues 1`] = `
Object {
  "addMany": [Function],
  "addOne": [Function],
  "getInitialState": [Function],
  "getSelectors": [Function],
  "removeAll": [Function],
  "removeMany": [Function],
  "removeOne": [Function],
  "selectId": [Function],
  "setAll": [Function],
  "sortComparer": false,
  "updateMany": [Function],
  "updateOne": [Function],
  "upsertMany": [Function],
  "upsertOne": [Function],
}
`;

exports[`entity adapters issues adapter returns generated selector for the issues adapter 1`] = `
Object {
  "selectAll": [Function],
  "selectById": [Function],
  "selectEntities": [Function],
  "selectIds": [Function],
  "selectTotal": [Function],
}
`;
